source-browser.html

```diff
@@ -182,6 +193,29 @@ <h1 id="loading" class="m1 center display-none">⌛</h1>
       }
 
       $(document).ready(() => {
+        $(window).on('scroll resize', function(){
+            const top = $(window).scrollTop();
+            const bottom = top + $(window).height();
+            TOPICS.forEach(topic => {
+              let {name} = topic;
+              const name_position = $(`#${name}`).position().top;
+              if (top < name_position && name_position < bottom) {
+                $(`#${name}-nav-link`).hide();
+              } else {
+                $(`#${name}-nav-link`).text(`${name} ${name_position > bottom ? '↓' : '↑'}`).show();
+              }
+            });
+          });
+        TOPICS.forEach(topic => {
```

Comment:
This entire block can be replaced with `$('.li').append(TOPICS.map(({name}) => ...`

Reply:
Also, why are you appending all the links to the same list element? There should be one list element (`<li>`) per entry inside the `<ul>`.

Reply:
I tried replacing TOPICS.forEach(topic => { as you recommended, but it didn't work.

I also tried to make one list element (each containing one /<a> element), but it didn't quite work.  I tried doing this:

`          TOPICS.map(({name}) => {
            $('#navbar').append(
			  $('<li></li>', {class: `${name} left`})
              $('#navbar li').append(
                $('<a class = "inline-block white center btn:hover">', {
                  id: `${name}-nav-link`,
                  href: `#${name}`
                }).text(name)
              );
		    );
         });`, which failed.  Do you have any suggestions?